editor/src/core/tailwind/tailwind.ts

```diff
@@ -0,0 +1,243 @@
+    if (Array.isArray(corePlugins)) {
+      // This means we have an explicit list of all allowed core plugins
+      return corePlugins.includes(PreflightKey)
+    } else {
+      // We have an object that enables or disables specific plugins - all unlisted plugins are enabled
+      const explicitDisabled = corePlugins[PreflightKey] === false // Tailwind doesn't use truthiness here
+      return !explicitDisabled
+    }
+  }
+
+  return true
+}
+
+function convertTailwindToTwindConfig(tailwindConfig: any): Configuration {
+  const preflightEnabled = enablesPreflight(tailwindConfig)
+
+  return {
+    ...tailwindConfig,
+    preflight: preflightEnabled,
+  }
+}
+
+function getTailwindConfig(
+  tailwindFile: ProjectFile | null,
+  requireFn: RequireFn,
+): Either<any, Configuration> {
+  if (tailwindFile != null && isTextFile(tailwindFile)) {
+    try {
+      const requireResult = requireFn('/', TailwindConfigPath)
+      if (requireResult?.default != null) {
+        const twindConfig = convertTailwindToTwindConfig(requireResult.default)
+        return right(twindConfig)
+      } else {
+        return left('Tailwind config contains no default export')
+      }
+    } catch (error) {
+      return left(error)
+    }
+  }
+
+  return left('Invalid or missing tailwind config file type')
+}
+
+function useGetTailwindConfigFile(projectContents: ProjectContentTreeRoot): ProjectFile | null {
+  return React.useMemo(() => getContentsTreeFileFromString(projectContents, TailwindConfigPath), [
+    projectContents,
+  ])
+}
+
+function useGetTailwindConfig(
+  projectContents: ProjectContentTreeRoot,
+  requireFn: RequireFn,
+): Configuration {
+  const tailwindConfigFile = useGetTailwindConfigFile(projectContents)
+  return React.useMemo(() => {
+    const maybeConfig = getTailwindConfig(tailwindConfigFile, requireFn)
+    if (isRight(maybeConfig)) {
+      return maybeConfig.value
+    } else {
+      return {}
+    }
+  }, [tailwindConfigFile, requireFn])
+}
+
+interface TwindInstance {
+  element: HTMLStyleElement
+  observer: TwindObserver
+}
+
+let twindInstance: TwindInstance | null = null
+
+function clearTwind() {
+  if (twindInstance != null) {
+    twindInstance.observer.disconnect()
+    twindInstance.element.parentNode?.removeChild(twindInstance.element)
+  }
+}
+
+const adjustRuleScope = memoize(
+  (rule: string, prefixSelector: string | null): string => {
+    if (prefixSelector == null) {
+      return rule
+    } else {
+      const splitOnBrace = rule.split('{')
+      const splitSelectors = splitOnBrace[0].split(',')
+      const scopedSelectors = splitSelectors.map((s) => {
+        const lowerCaseSelector = s.toLowerCase().trim()
+
+        if (
+          lowerCaseSelector === ':root' ||
+          lowerCaseSelector === 'html' ||
+          lowerCaseSelector === 'head'
+        ) {
+          return prefixSelector
+        } else if (lowerCaseSelector === 'body') {
+          return `${prefixSelector} > *`
```

Comment:
This is required because of the subtle difference between using the `html` vs the `body` selector. Tailwind uses both in the preflight, in one case attaching a specific `font-family` rule to the `html` selector, and then attaching `font-family: inherit` to the `body` selector.